src/lib.rs

```diff
@@ -102,6 +142,49 @@ impl SynapseVersion {
     }
 }
 
+/// The configuration for a single synapse container.
+/// We copy some fields from the main config because they
+/// are otherwise without the context we want to use them for, which
+/// is managing the docker container, not deserializing a homserver config.
+#[derive(Debug)]
+pub struct ContainerConfig {
+    /// The docker port configuration.
+    pub port_mapping: String,
+    /// The hostname of the synapse container.
+    pub hostname: String,
+    /// The name of a docker network to place the container in.
+    pub docker_network: Option<String>,
+    /// The URL that can be used to access this container from the host.
+    pub public_url: String,
+    /// The name of the synapse server.
+    pub server_name: String,
+}
+
+impl ContainerConfig {
+    pub fn from_mx_tester_config(config: &Config) -> ContainerConfig {
+        let homeserver_config = &config.homeserver_config;
+        ContainerConfig {
+            docker_network: config.docker_config.docker_network.clone(),
+            port_mapping: config.docker_config.port_mapping.clone(),
+            hostname: config.docker_config.hostname.clone(),
+            public_url: homeserver_config
+                .get(&serde_yaml::Value::from("public_baseurl"))
+                .unwrap_or_else(|| {
+                    panic!("Could not get the public_baseurl from the homserver_config")
```

Comment:
Let's try and stop panicking immediately, that will make writing unit tests harder.

Reply:
Nit: hom**e**server, here and in a few other places.